pkg/bootstrap/bootstrap.go

```diff
@@ -188,14 +188,32 @@ func (c *BootstrapClient) applyManifests(ctx context.Context, objs []unstructure
 		}
 
 		bar.Describe(fmt.Sprintf("Applying %v (%v)", obj.GetName(), obj.GetKind()))
-		if err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
-			_, err = c.client.Resource(mapping.Resource).Namespace(obj.GetNamespace()).
-				Apply(ctx, obj.GetName(), &obj, metav1.ApplyOptions{Force: true, FieldManager: "glasskube"})
-			return err
-		}); err != nil {
-			return err
+		if obj.GetKind() == "Job" {
+			options := metav1.DeletePropagationBackground
+			err := c.client.Resource(mapping.Resource).Namespace(obj.GetNamespace()).Delete(
+				ctx,
+				obj.GetName(),
+				metav1.DeleteOptions{PropagationPolicy: &options})
+			if err != nil {
+				return err
+			}
+			_, err = c.client.Resource(mapping.Resource).Namespace(obj.GetNamespace()).Apply(
+				ctx,
+				obj.GetName(),
+				&obj,
+				metav1.ApplyOptions{Force: true, FieldManager: "glasskube"})
+			if err != nil {
+				return err
+			}
+		} else {
+			if err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
+				_, err = c.client.Resource(mapping.Resource).Namespace(obj.GetNamespace()).
+					Apply(ctx, obj.GetName(), &obj, metav1.ApplyOptions{Force: true, FieldManager: "glasskube"})
+				return err
+			}); err != nil {
+				return err
+			}
 		}
```

Comment:
This part is basically the same as the `else` branch. It would be better to remove this part and lift the code out of the else branch.

Something like this:

```suggestion
			}
		}

		if err = retry.RetryOnConflict(retry.DefaultRetry, func() error {
			_, err = c.client.Resource(mapping.Resource).Namespace(obj.GetNamespace()).
				Apply(ctx, obj.GetName(), &obj, metav1.ApplyOptions{Force: true, FieldManager: "glasskube"})
			return err
		}); err != nil {
			return err
		}

```